src/main/kotlin/com/autonomousapps/tasks/ArtifactsReportTask.kt

```diff
@@ -94,7 +94,7 @@ abstract class ArtifactsReportTask : DefaultTask() {
         }
       }.filterToSet { art ->
         exclusions.none { ex ->
-          art.dependency.identifier == ex.identifier
+          art.dependency.identifier == ex.identifier && art.dependency.configurationName == ex.configurationName
```

Comment:
My intuition is that this is the wrong approach. (My intuition might be wrong.) I don't have advice right now, but I'll try to take a look when I get a chance.

Reply:
Hey @autonomousapps , do you by any chance have any news regarding this PR

Reply:
Thanks for the reminder. I will try to take a look soon. Basically spent the past month on-call at work, a bit burned out!

Reply:
I think I'm just not clear on what your goal is here. Could you file an issue? What problem are you running into that you're trying to solve?

Reply:
The issue is that a transitively used ABI dependency (NotificationSender parameter in MainApplication) is not part of the suggestions because the dependency is excluded from the locations (because the integrationTest is marked as not interesting, but it filters on the whole project instead of project + sourceSet).

Reply:
It would help if you could write it up a bit more concretely as an issue. Even pseudocode is helpful. Sample build.gradle with dependency declarations, sample source code where dependencies are being used.